### PR TITLE
Show how long the queue will last

### PR DESCRIPTION
**Pull request by tobidwest**
#49

### CODE REVIEW

File: common/page.tsx [hunk 1]
```diff
@@ -4,52 +4,93 @@ import { Queue } from "./components/Queue.tsx";
...
+            {/* @ts-ignore - uix doesn't support types? */}
+            <VideoPlayer queue={session.queue} code={code} />
+          </div>
+          <div class="flex items-center justify-end px-12 h-10 mb-4">
+            <ToggleThemeButton />
+          </div>
+
+          {current}
+          <div class="px-4 py-4 border-t border-black dark:border-white/20 mx-0 overflow-y-scroll flex-grow">
+            {always(() =>
```
Comment (flo-bit) [inline common/page.tsx:75]: the text returned here needs a dark mode text color

File: common/page.tsx [hunk 2]
```diff
@@ -4,52 +4,93 @@ import { Queue } from "./components/Queue.tsx";
...
+          </div>
+          <NowPlaying item={session.currentlyPlaying.$} />
+        </div>
+      );
+    } else {
+      return null;
+    }
+  });
+
+  function addDurations(duration1, duration2) {
```
Comment (flo-bit) [inline common/page.tsx:31]: this function gives me lots of warnings from my typescript thingy
- duration1 and duration2 are not typed
- all except the last `let` should be a `const`
- seconds probably shouldn't be type 'string' | 'number' (`String(n).padStart(1, '0')` may be interesting here)

Also I think it'd be good to move that to some other file, so we don't clutter up our main file, maybe a new `/common/helper.ts`, what do you think?

File: common/page.tsx [hunk 3]
```diff
@@ -4,52 +4,93 @@ import { Queue } from "./components/Queue.tsx";
...
+    let timeCounter = "0:00";
+    session.queue.forEach((item) => {
+      timeCounter = addDurations(timeCounter, item.duration);
+    });
+    return timeCounter;
+  });
+
+  loadInitialTheme();
+
+  window.timeLeft = timeLeft;
```
Comment (flo-bit) [inline common/page.tsx:53]: why are we doing this?

File: common/client.tsx
Comment (flo-bit) [inline common/client.tsx:1]: we really have to setup some common prettier settings ^^

is there any change except for formatting in this file? I can't find any

**Review by flo-bit — CHANGES_REQUESTED** [review]

File: common/client.tsx
Comment (tobidwest) [reply common/client.tsx:1]: no but it was indeed Prettier :)

File: common/page.tsx (hunk 3 above)
Comment (tobidwest) [reply common/page.tsx:53]: leftover from debugging, dropped that

**Comment by tobidwest** [comment]
See #eaa383293f152866202a1cd7e7b12902a3e52ab6

**Comment by tobidwest** [comment]
See eaa383293f152866202a1cd7e7b12902a3e52ab6

**Review by flo-bit — APPROVED** [review]
👍

**Pull request merged**